sdk/modelsrepository/Azure.Iot.ModelsRepository/src/ResolverClient.cs

```diff
@@ -0,0 +1,129 @@
+        /// <c>ResolverClientOptions</c> to configure resolution and client behavior.
+        /// </param>
+        public ResolverClient(ResolverClientOptions options) : this(new Uri(DefaultRepository), options) { }
+
+        /// <summary>
+        /// Initializes the <c>ResolverClient</c> with custom client <paramref name="options"/> while pointing to
+        /// a custom <paramref name="repositoryUri"/> for resolution.
+        /// </summary>
+        /// <param name="repositoryUri">
+        /// The model repository <c>Uri</c>. This can be a remote endpoint or local directory.
+        /// </param>
+        /// <param name="options">
+        /// <c>ResolverClientOptions</c> to configure resolution and client behavior.
+        /// </param>
+        public ResolverClient(Uri repositoryUri, ResolverClientOptions options)
+        {
+            _repositoryHandler = new RepositoryHandler(repositoryUri, options);
+        }
+
+        /// <summary>
+        /// Initializes the <c>ResolverClient</c> with default client options while pointing to
+        /// a custom <paramref name="repositoryUriStr"/> for resolution.
+        /// </summary>
+        /// <param name="repositoryUriStr">
+        /// The model repository <c>Uri</c> in string format. This can be a remote endpoint or local directory.
+        /// </param>
+        public ResolverClient(string repositoryUriStr) : this(repositoryUriStr, null) { }
+
+        /// <summary>
+        /// Initializes the <c>ResolverClient</c> with custom client <paramref name="options"/> while pointing to
+        /// a custom <paramref name="repositoryUriStr"/> for resolution.
+        /// </summary>
+        /// <param name="repositoryUriStr">
+        /// The model repository <c>Uri</c> in string format. This can be a remote endpoint or local directory.
+        /// </param>
+        /// <param name="options">
+        /// <c>ResolverClientOptions</c> to configure resolution and client behavior.
+        /// </param>
+        public ResolverClient(string repositoryUriStr, ResolverClientOptions options) : this(new Uri(repositoryUriStr), options) { }
+
+        /// <summary>
+        /// Resolves a model definition identified by <paramref name="dtmi"/> and optionally its dependencies.
+        /// </summary>
+        /// <returns>
+        /// An <c>IDictionary</c> containing the model definition(s) where the key is the dtmi
+        /// and the value is the raw model definition string.
+        /// </returns>
+        /// <exception cref="ResolverException">Thrown when a resolution failure occurs.</exception>
+        /// <param name="dtmi">A well-formed DTDL model Id. For example 'dtmi:com:example:Thermostat;1'.</param>
+        /// <param name="cancellationToken">The cancellationToken.</param>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Usage", "AZC0004:DO provide both asynchronous and synchronous variants for all service methods.", Justification = "<Pending>")]
```

Comment:
we need to discuss these items during our meeting with the SDK team today.

Reply:
What stops us from implementing both?

Reply:
Nothing, just wanted Paymaun to be included in the conversation before I start making API changes. This will come in at a later PR

Reply:
Which is much appreciated - I love this kind of collaboration, feedback and messaging.